File: tests/fixtures/ruby-syntax-family/host/lib/syntax_family/standards.rb
# frozen_string_literal: true

module SyntaxFamily
  module Standards
    def self.parse_invoice(value = 7)
      value
    end

    def self.handled_parse
      begin
        parse_invoice
      rescue StandardError
        0
      end
    end

    def self.unhandled_parse
      parse_invoice
    end

    CALL_SHAPED_STRING = "parse_invoice"
  end
end

File: tests/fixtures/c-syntax-cohort/host/src/standards.c
#include "cohort.h"

int parse_invoice(void)
{
    return 7;
}

int handled_parse(void)
{
    if (parse_invoice() > 0) {
        return 7;
    }
    return 0;
}

int unhandled_parse(void)
{
    return parse_invoice();
}

const char *call_shaped_string = "parse_invoice()";
